android/src/main/java/com/swmansion/rnscreens/utils/ScreenDummyLayoutHelper.kt

```diff
@@ -208,7 +213,11 @@ internal class ScreenDummyLayoutHelper(
         // scenarios when layout violates measured dimensions.
         coordinatorLayout.layout(0, 0, decorViewWidth, decorViewHeight)
 
-        val headerHeight = PixelUtil.toDIPFromPixel(appBarLayout.height.toFloat())
+        // Include the top inset to account for the extra padding manually applied to the CustomToolbar.
+        // This ensures the total height calculation of the AppBarLayout is accurate in edge-to-edge layouts.
+        val totalAppBarLayoutHeight = appBarLayout.height.toFloat() + topInset
+
+        val headerHeight = PixelUtil.toDIPFromPixel(totalAppBarLayoutHeight)
```

Comment:
Using words like "ensures" in this context is courageous 😄

Reply:
https://github.com/software-mansion/react-native-screens/pull/3442/commits/07101c685d6ef2bdad7b5319ec11b03b3f37053c